packages/core/test/insured-bridge/BridgePool.js

```diff
@@ -1431,4 +1429,100 @@ describe("BridgePool", () => {
       );
     });
   });
+  describe("Canonical bridge finalizing before insured bridge settlement edge cases", () => {
+    beforeEach(async function () {
+      await l1Token.methods.mint(liquidityProvider, initialPoolLiquidity).send({ from: owner });
+      await l1Token.methods.approve(bridgePool.options.address, MAX_UINT_VAL).send({ from: liquidityProvider });
+      await bridgePool.methods.addLiquidity(initialPoolLiquidity).send({ from: liquidityProvider });
+      await l1Token.methods.mint(relayer, totalRelayBond.muln(100)).send({ from: owner });
+      await l1Token.methods.approve(bridgePool.options.address, totalRelayBond.muln(100)).send({ from: relayer });
+    });
+    it("Exchange rate correctly handles the canonical bridge finalizing before insured relayer begins", async () => {
+      // Consider the edge case where a user deposits on L2 and no actions happen on L1. This might be due to their
+      // deposit fees being under priced (not picked up by a relayer). After a week their funds arrive on L1 via the
+      // canonical bridge. At this point, the transfer is relayed. The exchange rate should correctly deal with this
+      // without introducing a step in the rate at any point.
+
+      // Advance time by 1 week past the end of the of the L2->L1 liveness period.
+      await advanceTime(604800);
+      assert.equal((await bridgePool.methods.exchangeRateCurrent().call()).toString(), toWei("1"));
+
+      // Now, simulate the finalization of of the bridge action by the canonical bridge by minting tokens to the pool.
+      await l1Token.methods.mint(bridgePool.options.address, toWei("100")).send({ from: owner });
+
+      // The exchange rate should not have updated.
```

Comment:
Could you briefly explain how this is possible? Looking at the math, it was a little difficult for me to figure out.

I think it's because utilizedReserves goes _negative_, which drags down the numerator by the same amount of unexpected tokens dumped on the contract, but I want to make sure your intuition matches mine.

Reply:
yup! that maps exactly to why I think this works. It's all about the negative `utilizedReserves` that captures this perfectly. when we discussed this IRL we neglected this fact :)

Reply:
👌